playback-service/src/main/java/com/playback/PlaybackStore.java

```diff
@@ -0,0 +1,57 @@
+package com.playback;
+
+import com.playback.models.*;
+import com.playback.services.ContentSecurityPolicyService;
+import com.playback.services.CustomerInfoService;
+import com.playback.services.SteeringService;
+import com.playback.services.VideoMetadataService;
+import com.playback.standards.VideoResolution;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class PlaybackStore {
+    private final VideoMetadataService videoMetadataService;
+    private final CustomerInfoService customerInfoService;
+    private final ContentSecurityPolicyService contentSecurityPolicyService;
+    private final SteeringService steeringService;
+
+    public PlaybackStore(VideoMetadataService videoMetadataService, CustomerInfoService customerInfoService, ContentSecurityPolicyService contentSecurityPolicyService, SteeringService steeringService) {
+        this.videoMetadataService = videoMetadataService;
+        this.customerInfoService = customerInfoService;
+        this.contentSecurityPolicyService = contentSecurityPolicyService;
+        this.steeringService = steeringService;
+    }
+
+    public Optional<PlaybackManifest> request(String oauth, String title, Device device) {
```

Comment:
The current implementation does not signal to the device/customer why the video is not playable.  How might you update this to indicate different failure scenarios?

Reply:
We could add to the `PlaybackManifest` failure codes and reasons. Then the client could leverage these known codes/reasons to display a useful error message to the user through whatever device. Alternatively you could leverage exceptions to drive this behavior. 

We could also construct a new return type that would be composed with the `PlaybackManifest` itself if things are successful or an error code or `Exception` if things failed. That way the return type itself is responsible for conveying the success or failure rather than leveraging `Optional` which can only contain two states.